File: src/main/java/com/bemsa/scraper/models/GitUser.java
package com.bemsa.scraper.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.ObjectMapper;
import lombok.*;

import java.time.ZonedDateTime;
import java.util.List;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@JsonIgnoreProperties(ignoreUnknown = true)
@Builder
public final class GitUser {
    @NonNull
    private String login;

    @NonNull
    private String name;

    @NonNull
    @JsonProperty("avatar_url")
    private String avatarUrl;

    @NonNull
    private String location;

    private String email;

    @NonNull
    @JsonProperty("html_url")
    private String url;

    @NonNull
    @JsonProperty("created_at")
    private ZonedDateTime createdAt;

    @NonNull
    @Setter
    private List<GitRepo> repos;
}

File: src/main/kotlin/com/bemsa/scraper/models/GitUser.kt
package com.bemsa.scraper.models

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty
import java.time.ZonedDateTime

@JsonIgnoreProperties(ignoreUnknown = true)
data class GitUser (
    val login: String,
    val name: String,
    @JsonProperty("avatar_url") val avatarUrl: String,
    val location: String,
    val email: String?,
    @JsonProperty("html_url") val url: String,
    @JsonProperty("created_at") val createdAt: ZonedDateTime,
    var repos: MutableList<GitRepo>
)
